pkg/kv/kvserver/loqrecovery/recovery_env_test.go

```diff
@@ -712,52 +710,19 @@ func (e *quorumRecoveryEnv) parseStoresArg(
 	t *testing.T, d datadriven.TestData, defaultToAll bool,
 ) []roachpb.StoreID {
 	// Prepare replica info
-	var stores []roachpb.StoreID
-	if d.HasArg("stores") {
-		for _, arg := range d.CmdArgs {
-			if arg.Key == "stores" {
-				for _, id := range arg.Vals {
-					id, err := strconv.ParseInt(id, 10, 32)
-					if err != nil {
-						t.Fatalf("failed to parse store id: %v", err)
-					}
-					stores = append(stores, roachpb.StoreID(id))
-				}
-			}
-		}
-	} else {
-		if defaultToAll {
-			for id := range e.stores {
-				stores = append(stores, id)
-			}
-		} else {
-			stores = []roachpb.StoreID{}
-		}
+	stores, ok := dd.ScanArgOpt[[]roachpb.StoreID](t, &d, "stores")
+	if !ok && defaultToAll {
+		stores = slices.AppendSeq(stores, maps.Keys(e.stores))
+		slices.Sort(stores)
 	}
-	slices.Sort(stores)
```

Comment:
Perhaps we should still sort outside of the `if` block above so that user-provided input is still sorted? I'm not familiar with this test to know if it matters much.

Reply:
Done. It hadn't broken any tests, but done to preserve the old semantics.